integrations/micrometer/micrometer/src/main/java/io/helidon/integrations/micrometer/MeterRegistryFactory.java

```diff
@@ -237,12 +240,25 @@ Handler matchingHandler(ServerRequest serverRequest, ServerResponse serverRespon
                         .send(NO_MATCHING_REGISTRY_ERROR_MESSAGE));
     }
 
+    io.helidon.nima.webserver.http.Handler matchingHandler(io.helidon.nima.webserver.http.ServerRequest serverRequest,
+                                                           io.helidon.nima.webserver.http.ServerResponse serverResponse) {
+        return nimaRegistryEnrollments.stream()
+                .map(e -> e.handlerFn().apply(serverRequest))
+                .findFirst()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
```

Comment:
```suggestion
                .flatMap(Function.identity())
```

Reply:
Done differently and shortened + now it should do what seems was expected for it to do